test/configLoader.test.js

```diff
@@ -1,154 +1,211 @@
+  describe('configLoader', () => {
+    beforeAll(() => {
+      lookUpEnvironmentVar.mockImplementation(variableName => process.env[variableName]);
+      getEnvironmentVarPointers.mockReturnValue(yamlConfigPointers);
+      fs.readFileSync.mockImplementation(filePath => {
+        if(/(\.yml)|(\.yaml)/.test(filePath)) {
+          return rawYamlConfig;
+        } else if (/(\.json)/.test(filePath)) {
+          return rawJsonConfig;
+        }
+        return '';
+      });
+
+      // define valid env variables in the config file
+      process.env.MY_BIG_SECRET = 'no one can know';
+      process.env.NOUN_1 = 'excess';
+      process.env.NOUN_2 = 'riches';
 
-  describe('file errors', () => {
-    it('should throw if the file does not exist', () => {
-      fs.existsSync.mockReturnValue(false);
-      expect(() => {
-        configLoader('./myConfig.yml');
-      }).to.throwException("Supplied --path './myConfig.yml' doesn't exist");
+      // suppress console error output: .red isn't defined, so these aren't useful
+      jest.spyOn(log, 'error').mockImplementation(() => {});
     });
 
-    it('should throw if the file is in an unsupported format', () => {
-      expect(() => {
-        configLoader('myConfig.js');
-      }).to.throwException("myConfig.js is not a supported Kongfig config format");
+    beforeEach(() => {
+      fs.existsSync.mockReturnValue(true);
     });
-  });
 
-  describe('supported formats', () => {
-    it('should load YAML config files', () => {
-      let config;
-      ['config.yml', 'config.yaml'].forEach(filename => {
+    describe('file errors', () => {
+      it('should throw if the file does not exist', () => {
+        fs.existsSync.mockReturnValue(false);
         expect(() => {
-          config = configLoader('config.yml');
-        }).not.to.throwException();
+          configLoader('./myConfig.yml');
+        }).to.throwException("Supplied --path './myConfig.yml' doesn't exist");
+      });
 
-        expect('my_yaml_config' in config).to.be.ok();
-        expect(config.my_yaml_config.property).to.be('value');
+      it('should throw if the file is in an unsupported format', () => {
+        expect(() => {
+          configLoader('myConfig.js');
+        }).to.throwException("myConfig.js is not a supported Kongfig config format");
       });
     });
 
-    it('should load JSON config files', () => {
-      let config;
-      expect(() => {
-        config = configLoader('config.json');
-      }).not.to.throwException();
-      
-      expect('myJsonConfig' in config).to.be.ok();
-      expect(config.myJsonConfig.property).to.be('value');
-    });
-  });
+    describe('supported formats', () => {
+      it('should load YAML config files', () => {
+        let config;
+        ['config.yml', 'config.yaml'].forEach(filename => {
+          expect(() => {
+            config = configLoader('config.yml')[0];
+          }).not.to.throwException();
+
+          expect('my_yaml_config' in config).to.be.ok();
```

Comment:
`ok` just tests if the value is truthy, right? I can't seem to find the docs for it

Reply:
yup exactly